### ISSUE TITLE
Use Checkstyle checks to enforce usage of period in all message files 

### ISSUE THREAD

**Issue by suniti0804** [checkstyle/checkstyle#14007]
All the message properties files should contain period at the end of the message. Message in all the languages must be consistent with each other. 

Example - Message file of https://github.com/checkstyle/checkstyle/blob/cc5cb8e706aa51dfd211a97fe9d7e4a2369e12c5/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java

**Comment by romani** [comment]
I am ok, but we need to be careful as Japan and Chinese might use special character for this, not regular `.` ( ASCII).

**Comment by nrmancuso** [comment]
> Message in all the languages must be consistent with each other.

This is an entirely separate issue that we will never complete without massive efforts from native speakers.

Another consideration for this issue: change in message could technically be a breaking change for some users if they had suppressions based on message content.

**Comment by romani** [comment]
I am ok to not mark this as non breaking changes.

>Message in all the languages must be consistent with each other.

Please remove this request, they are written in way language require and good for humans to read.

I am ok to start update of code with `.` even without some enforcement by some existing Checks, we can do this later.

**Comment by romani** [comment]
@nrmancuso , please review and approve

**Comment by nrmancuso** [comment]
@romani please see the issue description, we have a fundamental misunderstanding. @suniti0804’s concern is about punctuation in Javadocs, not message property files.

@suniti0804 do we not have a JavadocStyleCheck module in our checkstyle-checks configuration file?

**Comment by romani** [comment]
@nrmancuso , issue title and description is rewritten, please review.

**Comment by suniti0804** [comment]
@nrmancuso Yes, we do have a JavadocStyleCheck module in our checkstyle-checks configuration file

```
 <module name="JavadocStyle">
      <property name="scope" value="public"/>
 </module>
```

**Comment by nrmancuso** [comment]
Ok, I am still a bit confused about what the actual issue is. I don't understand how JavadocStyle is going to help us to find violations. I think that we lost a lot of context for my initial response at https://github.com/checkstyle/checkstyle/issues/14007#issuecomment-1801918769 from editing the issue description so many times. Attention to my request:

> Please update your issue description using the bug report template to show an example of your proposed module and what you would expect.

I still don't see any proposal of how to enforce this, so I will share what I would expect to be in this issue description if I understand the issue correctly.

------------------

All the message properties files should contain a period at the end of the message, to be consistent.

Example: 
https://github.com/checkstyle/checkstyle/blob/8780d60e7da3c0260fae4d91880eb4fb4b13130b/src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages.properties#L8-L11

We could enforce this rule with something like the following module (might need MultilineRegexp for line-wrapped messages):

```bash
➜  src cat test.properties 
my.prop.1=This message is good, it ends with a period.
my.prop.2=This message is not good, it does not end with a period
my.prop.3=This message has a period in the middle. This is fine since there is one at the end.
my.prop.4=This message is not good. It has a period in the middle, but not the end

➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
     "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="RegexpSingleline">
      <property name="format" value=".*[^\.]$"/>
      <property name="maximum" value="0"/>
      <property name="message"
                value="All check messages must end with a period."/>
      <property name="fileExtensions" value="properties"/>
  </module>
</module>

➜  src java -jar checkstyle-10.12.3-all.jar -c config.xml test.properties
Starting audit...
[ERROR] /home/nick/IdeaProjects/tester/src/test.properties:2: All check messages must end with a period. [RegexpSingleline]
[ERROR] /home/nick/IdeaProjects/tester/src/test.properties:4: All check messages must end with a period. [RegexpSingleline]
Audit done.
Checkstyle ends with 2 errors.
```

Proof of necessity:
```
➜  checkstyle git:(master) ✗ git diff
diff --git a/config/checkstyle-checks.xml b/config/checkstyle-checks.xml
index 8c6fceefe..e72a5c511 100644
--- a/config/checkstyle-checks.xml
+++ b/config/checkstyle-checks.xml
@@ -36,6 +36,14 @@
   <module name="SuppressionFilter">
     <property name="file" value="${checkstyle.suppressions.file}"/>
   </module>
+
+  <module name="RegexpSingleline">
+    <property name="format" value=".*[^\.]$"/>
+    <property name="maximum" value="0"/>
+    <property name="message"
+              value="All check messages must end with a period."/>
+    <property name="fileExtensions" value="properties"/>
+  </module>
   <!-- Tone down the checking for test code -->
   <module name="SuppressionSingleFilter">
     <property name="checks" value="JavadocPackage"/>

➜  checkstyle git:(master) ✗ mvn clean compile antrun:run@ant-phase-verify

[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ------------------< com.puppycrawl.tools:checkstyle >-------------------
[INFO] Building checkstyle 10.12.5-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.3.2:clean (default-clean) @ checkstyle ---
[INFO] Deleting /home/nick/IdeaProjects/checkstyle/target
[INFO] 
[INFO] --- tidy-maven-plugin:1.2.0:check (validate) @ checkstyle ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.4.1:enforce (enforce-maven) @ checkstyle ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.version.RequireMavenVersion passed
[INFO] 
[INFO] --- maven-enforcer-plugin:3.4.1:enforce (enforce-versions) @ checkstyle ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion passed
[INFO] Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion passed
[INFO] Rule 2: org.apache.maven.enforcer.rules.dependency.DependencyConvergence passed
[INFO] 
[INFO] --- xml-maven-plugin:1.1.0:check-format (default) @ checkstyle ---
[INFO] 
[INFO] --- antlr4-maven-plugin:4.13.1:antlr4 (default) @ checkstyle ---
[INFO] ANTLR 4: Processing source directory /home/nick/IdeaProjects/checkstyle/src/main/resources
[INFO] Processing grammar: com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageLexer.g4
[INFO] Processing grammar: com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageParser.g4
[INFO] Processing grammar: com/puppycrawl/tools/checkstyle/grammar/javadoc/JavadocLexer.g4
[INFO] Processing grammar: com/puppycrawl/tools/checkstyle/grammar/javadoc/JavadocParser.g4
[INFO] 
[INFO] --- build-helper-maven-plugin:3.4.0:add-source (add-source) @ checkstyle ---
[INFO] Source directory: /home/nick/IdeaProjects/checkstyle/target/generated-sources/antlr added.
[INFO] 
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ checkstyle ---
[INFO] Copying 402 resources from src/main/resources to target/classes
[INFO] 
[INFO] --- build-helper-maven-plugin:3.4.0:add-test-source (add-test-source) @ checkstyle ---
[INFO] Test Source directory: /home/nick/IdeaProjects/checkstyle/src/test/resources added.
[INFO] 
[INFO] --- build-helper-maven-plugin:3.4.0:add-test-source (add-it-test-source) @ checkstyle ---
[INFO] Test Source directory: /home/nick/IdeaProjects/checkstyle/src/it/java added.
...

[ERROR] [checkstyle] [ERROR] /home/nick/IdeaProjects/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/messages_zh.properties:18: All check messages must end with a period. [RegexpSingleline]
[ERROR] [checkstyle] [ERROR] /home/nick/IdeaProjects/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/meta/messages_ja.properties:1: All check messages must end with a period. [RegexpSingleline]
[ERROR] [checkstyle] [ERROR] /home/nick/IdeaProjects/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/meta/messages_zh.properties:1: All check messages must end with a period. [RegexpSingleline]
[INFO]      [echo] Checkstyle finished (checkstyle-checks.xml) : 11/11/2023 01:24:35 AM
[INFO]      [echo] Checkstyle started (checkstyle-non-main-files-checks.xml): 11/11/2023 01:23:51 AM
[INFO] [checkstyle] Running Checkstyle  on 4223 files
[INFO]      [echo] Checkstyle finished (checkstyle-non-main-files-checks.xml): 11/11/2023 01:24:35 AM
[INFO]      [echo]     
[INFO]      [echo] Checkstyle started (checkstyle-resources-checks.xml): 11/11/2023 01:23:51 AM
[INFO] [checkstyle] Running Checkstyle  on 3482 files
[INFO]      [echo] Checkstyle finished (checkstyle-resources-checks.xml): 11/11/2023 01:24:35 AM
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:47 min
[INFO] Finished at: 2023-11-11T01:25:29-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (ant-phase-verify) on project checkstyle: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] /home/nick/IdeaProjects/checkstyle/config/ant-phase-verify.xml:183: Checkstyle failed: Got 788 errors and 0 warnings.
[ERROR] around Ant part ...<ant antfile="config/ant-phase-verify.xml" />... @ 6:50 in /home/nick/IdeaProjects/checkstyle/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
➜  checkstyle git:(master) ✗ 

```

@suniti0804 please confirm that I understand the issue correctly.

**Comment by suniti0804** [comment]
@nrmancuso thank you so much for framing the issue this way. this does captures everything about the issue correctly.
